src/app/knowledgeapi.service.spec.ts

```diff
@@ -1,25 +1,75 @@
-import { TestBed, inject } from '@angular/core/testing';
+/* tslint:disable:no-unused-variable */
+
+import { TestBed, inject, async, fakeAsync, tick } from '@angular/core/testing';
+import { MockBackend, MockConnection } from '@angular/http/testing';
+import { Http, BaseRequestOptions, RequestMethod, Response, ResponseOptions, HttpModule, JsonpModule } from '@angular/http';
+import { StoreModule } from '@ngrx/store';
+import { reducer } from './reducers/index';
 
 import { KnowledgeapiService } from './knowledgeapi.service';
-import {HttpModule, JsonpModule} from "@angular/http";
-import {reducer} from "./reducers/index";
-import {StoreModule} from "@ngrx/store";
-import {StoreDevtoolsModule} from "@ngrx/store-devtools";
 
-describe('KnowledgeapiService', () => {
+import { MockKnowledgeApi } from './shared/mock-backend/knowledge.mock';
+
+const mockHttp_provider = {
+  provide: Http,
+  deps: [MockBackend, BaseRequestOptions],
+  useFactory: (backend: MockBackend, options: BaseRequestOptions) => {
+    return new Http(backend, options);
+  }
+}
+
+describe('Service: KnowledgeapiService', () => {
+  let service: KnowledgeapiService = null;
+  let backend: MockBackend = null;
+
   beforeEach(() => {
     TestBed.configureTestingModule({
+      providers: [
+        KnowledgeapiService,
+        MockBackend,
+        BaseRequestOptions,
+        mockHttp_provider,
+      ],
       imports: [
         HttpModule,
         JsonpModule,
-        StoreModule.provideStore(reducer),
-        StoreDevtoolsModule.instrumentOnlyWithExtension(),
-      ],
-      providers: [KnowledgeapiService]
+        StoreModule.provideStore(reducer)
+      ]
     });
   });
 
-  it('should ...', inject([KnowledgeapiService], (service: KnowledgeapiService) => {
-    expect(service).toBeTruthy();
+  beforeEach(inject([KnowledgeapiService, MockBackend], (knowledgeService: KnowledgeapiService, mockBackend: MockBackend) => {
+    service = knowledgeService;
+    backend = mockBackend;
   }));
+
+  const searchquery = 'Berlin';
+  const _queryResult = MockKnowledgeApi;
+
+  it('should create an instance KnowledgeapiService',
+    inject([KnowledgeapiService, MockBackend], () => {
+      expect(service).toBeTruthy();
+    })
+  );
+
+  it('should call knowledge service API and return the result', () => {
+    backend.connections.subscribe((connection: MockConnection) => {
+      const options = new ResponseOptions({
+        body: JSON.stringify(MockKnowledgeApi)
+      });
+    
+      connection.mockRespond(new Response(options));
+      expect(connection.request.method).toEqual(RequestMethod.Get);
+      expect(connection.request.url).toBe(
+        `http://lookup.dbpedia.org/api/search/KeywordSearch` +
```

Comment:
the error you have raised is being called from here, so might be you '&' before QueryString